src/asmcnc/apps/drywall_cutter_app/widget_drywall_shape_display.py

```diff
@@ -298,14 +328,69 @@ def rotation_required(self):
         else:
             return False
 
-    def poll_position(self, dt):
-        current_x = round(abs(self.m.mpos_x()), 2)
-        current_y = round(abs(self.m.mpos_y()), 2)
+    def check_datum_and_extents(self, dt):
+        # Maths from Ed, documented here https://docs.google.com/spreadsheets/d/1X37CWF8bsXeC0dY-HsbwBu_QR6N510V-5aPTnxwIR6I/edit#gid=677510108
+        current_x = round(self.m.x_wco() + (self.m.get_dollar_setting(130) - self.m.limit_switch_safety_distance) - self.m.laser_offset_tool_clearance_to_access_edge_of_sheet, 2)
+        current_y = round(self.m.y_wco() + (self.m.get_dollar_setting(131) - self.m.limit_switch_safety_distance) - (self.m.get_dollar_setting(27) - self.m.limit_switch_safety_distance), 2)
         self.x_datum_label.text = 'X: ' + str(current_x)
         self.y_datum_label.text = 'Y: ' + str(current_y)
 
-        if self.dwt_config.active_config.datum_position.x != current_x:
-            self.dwt_config.active_config.datum_position.x = current_x
+        # Account for cutter size
+        cutter_radius = self.dwt_config.active_cutter.diameter / 2
+        if self.dwt_config.active_config.toolpath_offset == 'inside':
+            tool_offset_value = -cutter_radius
+        elif self.dwt_config.active_config.toolpath_offset == 'outside':
+            tool_offset_value = cutter_radius
+        else:
+            tool_offset_value = 0
+
+        # Calculate shape's extent from datum using shape type and input dimensions
+        current_shape = self.dwt_config.active_config.shape_type.lower()
+        if current_shape == 'circle':
+            x_min = y_min = -(float(self.d_input.text or 0) / 2) - tool_offset_value
+            x_dim = y_dim = (float(self.d_input.text or 0) / 2) + tool_offset_value
+        elif current_shape in ['square', 'rectangle']:
+            x_min = y_min = -tool_offset_value
+            y_dim = float(self.y_input.text or 0) + tool_offset_value
+            # As square only uses y input it needs a separate condition
+            if current_shape == 'square':
+                x_dim = y_dim
+            elif current_shape == 'rectangle':
```

Comment:
I think this class could benefit from an enum for the shape, tool path offset and any other config property. I will add a ticket to Jira for myself.

`class ShapeEnum(Enum):
    RECTANGLE = 'rectangle'
    CIRCLE = 'circle'`